akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -63,69 +150,66 @@ private[akka] class AffinityPool(
   // indicates the current state of the pool
   @volatile final private var poolState: PoolState = Running
 
-  private final val workQueues = Array.fill(parallelism)(new BoundedTaskQueue(affinityGroupSize))
-  private final val workers = mutable.Set[ThreadPoolWorker]()
+  private final val workQueues = Array.fill(parallelism)(new BoundedAffinityTaskQueue(affinityGroupSize))
+  private final val workers = mutable.Set[AffinityPoolWorker]()
 
-  // a counter that gets incremented every time a task is queued
-  private val executionCounter: AtomicInteger = new AtomicInteger(0)
   // maps a runnable to an index of a worker queue
-  private val runnableToWorkerQueueIndex = new AtomicReference(ImmutableIntMap.empty)
+  private[this] final val runnableToWorkerQueueIndex = new AtomicReference(ImmutableIntMap.empty)
 
-  private def locked[T](l: Lock)(body: ⇒ T) = {
-    l.lock()
+  @inline private def locked[T](body: ⇒ T): T = {
```

Comment:
Related news, hoping to try out `-opt-inline-from` https://contributors.scala-lang.org/t/upcoming-scala-releases-2-12-3-and-2-13-0-m1 this week or weekend to see how well it works, we can finally actually use the optimiser with that option I think